src/data/pi.lean

```diff
@@ -115,12 +115,18 @@ function.update_injective _ i
 end
 end pi
 
-section extend
-
 namespace function
-
 variables {α β γ : Type*}
 
+lemma add_comp [has_add γ] {h : α → β} {f g : β → γ} : (f + g) ∘ h = f ∘ h + g ∘ h := rfl
+lemma sub_comp [has_sub γ] {h : α → β} {f g : β → γ} : (f - g) ∘ h = f ∘ h - g ∘ h := rfl
+lemma mul_comp [has_mul γ] {h : α → β} {f g : β → γ} : (f * g) ∘ h = f ∘ h * g ∘ h := rfl
+lemma div_comp [has_div γ] {h : α → β} {f g : β → γ} : (f / g) ∘ h = f ∘ h / g ∘ h := rfl
+lemma neg_comp [has_neg γ] {g : α → β} {f : β → γ} : (-f) ∘ g = - f ∘ g := rfl
+lemma inv_comp [has_inv γ] {g : α → β} {f : β → γ} : f⁻¹ ∘ g = (f ∘ g)⁻¹ := rfl
```

Comment:
`f`, `g`, and `h` should be explicit here IMO.

I also think these would make more sense next to `one_apply`, `one_def`, etc.

Reply:
Oh, and don't forget `to_additive`.

Reply:
`pi.one_comp` has its argument implicit and is a simp lemma. I don't know what you want to do with that.